### ISSUE TITLE
Error: Storybook preview hooks can only be called inside decorators and story functions

### ISSUE THREAD

**Issue by k1350** [stevensacks/storybook-react-i18next#25]
I got the following error in v2.0.4 or later when switching locale with Next.js.

```
Storybook preview hooks can only be called inside decorators and story functions.
Error: Storybook preview hooks can only be called inside decorators and story functions.
    at invalidHooksError (http://localhost:6006/vendors-node_modules_babel_runtime_helpers_esm_defineProperty_js-node_modules_storybook_nextj-5100d7.iframe.bundle.js:698:5553)
    at useStoryContext (http://localhost:6006/vendors-node_modules_babel_runtime_helpers_esm_defineProperty_js-node_modules_storybook_nextj-5100d7.iframe.bundle.js:700:2227)
    at useGlobals (http://localhost:6006/vendors-node_modules_babel_runtime_helpers_esm_defineProperty_js-node_modules_storybook_nextj-5100d7.iframe.bundle.js:700:2914)
    at c (http://localhost:6006/vendors-node_modules_babel_runtime_helpers_esm_defineProperty_js-node_modules_storybook_nextj-5100d7.iframe.bundle.js:75238:528)
    at hookified (http://localhost:6006/sb-preview/runtime.js:7:17226)
    at http://localhost:6006/sb-preview/runtime.js:34:1451
    at http://localhost:6006/sb-preview/runtime.js:34:1912
    at renderWithHooks (http://localhost:6006/vendors-node_modules_babel_runtime_helpers_esm_defineProperty_js-node_modules_storybook_nextj-5100d7.iframe.bundle.js:55765:18)
    at updateFunctionComponent (http://localhost:6006/vendors-node_modules_babel_runtime_helpers_esm_defineProperty_js-node_modules_storybook_nextj-5100d7.iframe.bundle.js:59043:20)
    at beginWork (http://localhost:6006/vendors-node_modules_babel_runtime_helpers_esm_defineProperty_js-node_modules_storybook_nextj-5100d7.iframe.bundle.js:61056:16)
```

It works fine in v2.0.3.
Is there any workaround for this?

versions
```json
  "dependencies": {
    "@types/node": "20.4.1",
    "@types/react": "18.2.14",
    "@types/react-dom": "18.2.6",
    "eslint": "8.44.0",
    "eslint-config-next": "13.4.9",
    "i18next": "^23.2.8",
    "next": "13.4.9",
    "next-i18next": "^14.0.0",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-i18next": "^13.0.2",
    "typescript": "5.1.6"
  },
  "devDependencies": {
    "@storybook/nextjs": "^7.0.26",
    "storybook": "^7.0.26",
    "storybook-react-i18next": "^2.0.6"
  }
```

- npm -v
    - 9.5.1
- node -v
    - v18.16.0
- Minimum reproduction
    - https://github.com/k1350/storybook-react-i18next_sample

**Comment by theRuslan** [comment]
I have the same issue.
[k1350](https://github.com/k1350) did you find any solution?

**Comment by theRuslan** [comment]
> I downgraded storybook-react-i18next to v2.0.3.

Ok, I see, thanks.
I ended up with [this manual integration](https://storybook.js.org/recipes/react-i18next) without addon just with decorator and it works.

**Comment by stevensacks** [comment]
The issue seems to be with the function call `story(context)` vs the component `<Story />`. I will change this and release soon.

**Comment by stevensacks** [comment]
This is a strange one. When I use `story(context)` there's no error. When I use `<Story {...context />` I get the error you see.

I've changed the `useState` in the decorator to properly use the `@storybook/client-api` one instead of React.

I have released it as 2.0.8. Please let me know if this fixes the issue for you.
